Automattic/woocommerce-payments
# Enable ECE for shipping subscriptions with free trial and sign up fee

Part of https://github.com/Automattic/woocommerce-payments/issues/9771

#### Changes proposed in this Pull Request

Enables ECE for subscriptions that require shipping with a free trial and sign-up fee.

#### Testing instructions


> [!NOTE]  
> In order to test this PR with the changes in `woocommerce-subscriptions-core` ([ref](https://github.com/Automattic/woocommerce-subscriptions-core/pull/746)) you'd need to clone this repo and symlink the repo to your local `subscriptions-core` directory inside the WooCommerce Subscriptions plugin:
>```
>woopayments/wp-content/plugins/woocommerce-subscriptions » ll vendor/woocommerce
>lrwxr-xr-x 56 2024-11-05 15:28 subscriptions-core -> >/Users/asumaran/Developer/woocommerce-subscriptions-core
>```
> run `npm install` and `composer install` if necessary.



* Switch to this branch
* Ensure you are using the [PR](https://github.com/Automattic/woocommerce-subscriptions-core/pull/746) from `woocommerce-subscriptions-core`
* For reference, case matrix is [here](https://github.com/Automattic/woocommerce-payments/issues/9771#issuecomment-2518829514).

**Verify ECE is enabled for case 4**

- Create a simple subscription with a free trial and a sign up fee.
- Go to the product page.
- Verify ECE is enabled and can checkout without issues.

**Verify ECE is enabled for case 12**

- Create a variable subscription with at least one variation with only the following options:
  - With Free trial
  - With a Sign up fee.
- Go to the product page.
- Verify ECE is enabled and can checkout without issues.

**Verify ECE is not enabled for Case 3**

- Create a simple subscription with a free trial (do not add a sign up fee).
- Go to the product page.
- Verify ECE is not displayed.

**Verify ECE is not enabled for Case 11**

- Create a variable subscription with at least one variation with only the following options:
  - Free trial.
- Go to the product page.
- Verify ECE is hidden after selecting the variation.

-------------------


**Post merge**
